app/controllers/additionalInformation/index/AdditionalInformationTypeController.scala

```diff
@@ -0,0 +1,96 @@
+ * limitations under the License.
+ */
+
+package controllers.additionalInformation.index
+
+import config.PhaseConfig
+import controllers.actions._
+import controllers.{NavigatorOps, SettableOps, SettableOpsRunner}
+import forms.SelectableFormProvider
+import models.reference.additionalInformation.AdditionalInformationCode
+import models.requests.DataRequest
+import models.{Index, LocalReferenceNumber, Mode, SelectableList}
+import navigation.{TransportNavigatorProvider, UserAnswersNavigator}
+import pages.additionalInformation.index.AdditionalInformationTypePage
+import pages.preRequisites.ItemsDestinationCountryInCL009Page
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import services.AdditionalInformationService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.additionalInformation.index.AdditionalInformationTypeView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class AdditionalInformationTypeController @Inject() (
+  override val messagesApi: MessagesApi,
+  implicit val sessionRepository: SessionRepository,
+  navigatorProvider: TransportNavigatorProvider,
+  actions: Actions,
+  formProvider: SelectableFormProvider,
+  service: AdditionalInformationService,
+  val controllerComponents: MessagesControllerComponents,
+  view: AdditionalInformationTypeView
+)(implicit ec: ExecutionContext, phaseConfig: PhaseConfig)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad(additionalInformationIndex: Index, lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions.requireData(lrn).async {
+    implicit request =>
+      service.getAdditionalInformationCodes().map {
+        additionalInformationList =>
+          val (form, list) = getFormAndUpdatedCodeList(request, additionalInformationList)
+          val preparedForm = request.userAnswers.get(AdditionalInformationTypePage(additionalInformationIndex)) match {
+            case None        => form
+            case Some(value) => form.fill(value)
+          }
+
+          Ok(view(preparedForm, lrn, list.values, mode, additionalInformationIndex))
+      }
+  }
+
+  private def getFormAndUpdatedCodeList(request: DataRequest[AnyContent], additionalInformationList: SelectableList[AdditionalInformationCode]) =
+    request.userAnswers.get(ItemsDestinationCountryInCL009Page) match {
+      case Some(true) =>
+        val updatedList = SelectableList.apply(
+          additionalInformationList.values.filter(
+            code => !code.value.equals("30600")
+          )
+        )
+        (formProvider("additionalInformation.index.additionalInformationType", updatedList), updatedList)
+
+      case _ =>
+        (formProvider("additionalInformation.index.additionalInformationType", additionalInformationList), additionalInformationList)
+    }
```

Comment:
Could move this def to above the onPageLoad method and also refactor  to use filterNot, eg:
 private def getFormAndUpdatedCodeList(request: DataRequest[AnyContent], additionalInformationList: SelectableList[AdditionalInformationCode]) =
    request.userAnswers.get(ItemsDestinationCountryInCL009Page) match {
      case Some(true) =>
        val updatedList = SelectableList.apply(additionalInformationList.values.filterNot(_.value.equals("30600")))
        (formProvider("additionalInformation.index.additionalInformationType", updatedList), updatedList)
      case _ => (formProvider("additionalInformation.index.additionalInformationType", additionalInformationList), additionalInformationList)
    }

Reply:
@prashanthmamidi  Move to service layer?